src/test/java/com/hotels/beeju/extensions/ThriftHiveMetaStoreJUnitExtensionTest.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (C) 2015-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju.extensions;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.AlreadyExistsException;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.InvalidObjectException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.hotels.beeju.ThriftHiveMetaStoreJUnitRule;
+import com.hotels.beeju.ThriftHiveMetaStoreJUnitRuleTest;
+
+public class ThriftHiveMetaStoreJUnitExtensionTest implements BeforeEachCallback {
+
+  @RegisterExtension
+  ThriftHiveMetaStoreJUnitExtension hiveDefaultName = new ThriftHiveMetaStoreJUnitExtension();
+
+  @RegisterExtension
+  ThriftHiveMetaStoreJUnitExtension hiveCustomName = new ThriftHiveMetaStoreJUnitExtension("my_test_database");
+
+  private static File defaultTempRoot;
+  private static File customTempRoot;
+
+  @Override
+  public void beforeEach(ExtensionContext context){
+    defaultTempRoot = hiveDefaultName.getTempDirectory();
+    assertTrue(defaultTempRoot.exists());
+    customTempRoot = hiveCustomName.getTempDirectory();
+    assertTrue(customTempRoot.exists());
+  }
+
+  @Test
+  public void hiveDefaultName() throws Exception {
+    assertRuleInitialised(hiveDefaultName);
+  }
+
+  @Test
+  public void hiveCustomName() throws Exception {
+    assertRuleInitialised(hiveCustomName);
+  }
+
+  private static void assertRuleInitialised(ThriftHiveMetaStoreJUnitExtension hive) throws Exception {
+    String databaseName = hive.databaseName();
+
+    Database database = hive.client().getDatabase(databaseName);
+
+    assertThat(database.getName(), is(databaseName));
```

Comment:
This is more of a personal preference, but I wouldn't split the declaration of `databaseName` and of `database` from the assert for the database names. To me they should be together because they're about the same thing.

I would also add a new line between line 81 and 83.

Like I said, this is personal preference, so take it or leave it 🙂